### PR TITLE
[Ops][BugFix] Disable fused MC2 under speculative decoding to prevent logit corruption

### PR DESCRIPTION
**Pull request by ppppangu**
### What this does
On Ascend A3, `select_moe_comm_method` routes MoE comm to the fused `dispatch_ffn_combine` op (FUSED_MC2). Under speculative decoding (e.g. **DeepSeek-V4 + MTP**, `num_speculative_tokens=1`) this op is handed a tiny/ragged token shape and corrupts the model's **first-decode-step logits**, yielding intermittent **empty / malformed `<｜begin▁of▁sentence｜>` first responses** (gh #9170).

The current guard `dispatch_ffn_combine_enable = ... and (not is_draft_model)` only covers the **draft** forward. The corruption appears on the **target** model's verify-step forward, which runs with `is_draft_model=False`, so the fused op is still selected there.

This PR additionally gates on `vllm_config.speculative_config`, disabling fused MC2 for **both** the draft and the target-verify forwards whenever speculative decoding is configured (routing to MC2 / ALLTOALL). **Non-speculative runs are bit-identical** — zero behavior change when `speculative_config is None`. Under speculative decoding the MoE comm falls back to the decomposed MC2/ALLTOALL path instead of the fused op; `dispatch_ffn_combine` was already unusable for these tiny/ragged shapes (it is precisely what corrupts them), and MTP's own decode speedup dominates, so there is no net throughput regression for spec-decode either.

### Root cause (kernel detail in #9170)
`dispatch_ffn_combine` (AscendC) targets a dense, large decode batch and never zero-initializes its output / combine staging (sized by a fixed `max_output_size=65536`). On the tiny/ragged spec shape (~2 query positions/seq split unevenly across EP ranks, so a rank's local token count can be `< AIV core count`), the output-assembly tiling `unpermute/moe_token_unpermute_tiling.h:30-34` computes `tokens_core_length = outTokens/coreNum = 0` then divides by it (`0/0`), corrupting the loop bounds that write per-token output; under-filled rows are read from never-initialized HBM → NaN/garbage logits. A proper kernel fix (min-1 tiling guard + output zero-init) belongs in the op and touches the hot non-spec MoE path; this PR is the safe, side-effect-free control-plane mitigation.

### Validation
DeepSeek-V4-Flash W8A8 + MTP (`num_speculative_tokens=1`), 910C A3, TP4×DP4 + EP16: with fused MC2 enabled the first agent turn intermittently returns empty / `<｜begin▁of▁sentence｜>`-only output; setting `VLLM_ASCEND_ENABLE_FUSED_MC2=0` eliminates it (spec-decode sub-agents then complete normally). This PR makes that automatic under `speculative_config`.

Related to #9170 (mitigates the MTP/speculative trigger; the underlying kernel defect stays tracked there).

- vLLM version: v0.25.1
- vLLM main: https://github.com/vllm-project/vllm/commit/fe784ff22e630a31fd798f392b01e0a75c18f047

### CODE REVIEW

**Comment by ppppangu** [comment]
@wangxiyuan friendly ping 🙏 Rebased onto latest `main`, fixed the title prefix, and addressed the review note (switched to `getattr(vllm_config, "speculative_config", None)`). CI is green now — ready for review. This guards `dispatch_ffn_combine` off under speculative decoding, including the *target* verify-forward (which runs with `is_draft_model=False`), fixing the #9170 logit corruption (empty/garbage first-decode responses). Complementary to #9992, which hardens the op itself via zero-init. Thanks!

**Comment by fourierr** [comment]
在A3上关闭了fused_mc2，还是出现了这个问题，启动命令如下
export OMP_PROC_BIND=false
export OMP_NUM_THREADS=10
export PYTORCH_NPU_ALLOC_CONF=expandable_segments:True
export HCCL_BUFFSIZE=1024
export TASK_QUEUE_ENABLE=1
export HCCL_OP_EXPANSION_MODE="AIV"
export ENABLE_TRACE_LOGGING=true

vllm serve /DeepSeek-V4-Flash-w4a8-mtp/ \
    --max-model-len 262144 \
    --max-num-batched-tokens 16384 \
    --served-model-name illm \
    --gpu-memory-utilization 0.9 \
    --max-num-seqs 32 \
    --data-parallel-size 1 \
    --tensor-parallel-size 8 \
    --enable-expert-parallel \
    --tokenizer-mode deepseek_v4 \
    --tool-call-parser deepseek_v4 \
    --enable-auto-tool-choice \
    --reasoning-parser deepseek_v4 \
    --model-loader-extra-config='{"enable_multithread_load": true, "num_threads": 128}' \
    --quantization ascend \
    --port 8080 \
    --block-size 32 \
    --speculative-config '{"num_speculative_tokens": 3,"method": "mtp"}' \
    --compilation-config '{"cudagraph_mode": "FULL_DECODE_ONLY"}' \
    --additional-config '
    {"ascend_compilation_config":{
        "enable_npugraph_ex":true,
        "enable_static_kernel":false
        },
    "enable_cpu_binding": true,
    "enable_flashcomm1": true,
    "multistream_overlap_shared_expert": true}'
